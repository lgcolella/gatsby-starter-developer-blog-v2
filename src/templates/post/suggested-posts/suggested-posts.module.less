@import (reference) '../../../style/index.less';

.content {
    h3 {
        font-size: @font-size * 1.8;
        margin-bottom: @margin-zero;
    }
    p {
        font-size: @font-size * .9;
    }
}