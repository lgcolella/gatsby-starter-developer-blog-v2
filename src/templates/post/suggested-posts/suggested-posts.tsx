/* Vendor imports */
import React from 'react'
import PropTypes from 'prop-types'
/* App imports */
import PostList from '../../../components/post-list'
import { ChildImageSharpFluid, AvailableTag } from '../../../types'
const style = require('./suggested-posts.module.less')

const SuggestedPosts = ({ posts }: Props) => (
  <div>
    <PostList posts={posts} contentClassName={style.content} />
  </div>
)

SuggestedPosts.propTypes = {
  posts: PropTypes.arrayOf(
    PropTypes.shape({
      edge: PropTypes.shape({
        node: PropTypes.object,
      }),
    })
  ).isRequired,
}

export interface Props {
  posts: Array<{
    node: {
      fileAbsolutePath: string
      frontmatter: {
        path: string
        title: string
        tags: AvailableTag[]
        excerpt: string
        cover: {
          childImageSharp: ChildImageSharpFluid
        }
      }
    }
  }>
}

export default SuggestedPosts
