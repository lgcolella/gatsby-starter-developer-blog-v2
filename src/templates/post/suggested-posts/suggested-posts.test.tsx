/* Vendor imports */
import React from 'react'
import { shallow } from 'enzyme'
/* App imports */
import SuggestedPosts from './index'
import { Props } from './suggested-posts'
import { ImageSharpFluid } from '../../../../tests/mocks'

describe('Post > Suggsted Posts component', () => {
  const posts: Props['posts'] = [
    // Article #1
    {
      node: {
        fileAbsolutePath: 'C:/Documents/path-to-the-article-1',
        frontmatter: {
          title: 'Title of article #1',
          path: 'path-to-the-article-1',
          tags: ['angular', 'javascript', 'laravel'],
          excerpt: 'an excertp for the article #1',
          cover: {
            childImageSharp: ImageSharpFluid,
          },
        },
      },
    },
    // Article #2
    {
      node: {
        fileAbsolutePath: 'C:/Documents/path-to-the-article-2',
        frontmatter: {
          title: 'Title of article #2',
          path: 'path-to-the-article-2',
          tags: ['nodejs', 'angular'],
          excerpt: 'an excertp for the article #2',
          cover: {
            childImageSharp: ImageSharpFluid,
          },
        },
      },
    },
  ]

  test('matches snapshot', () => {
    let wrapper = shallow(<SuggestedPosts posts={posts} />)

    expect(wrapper).toMatchSnapshot()
  })
})
