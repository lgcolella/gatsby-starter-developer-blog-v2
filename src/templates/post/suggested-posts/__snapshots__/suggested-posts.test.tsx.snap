// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post > Suggsted Posts component matches snapshot 1`] = `
<div>
  <PostList
    contentClassName="content"
    posts={
      Array [
        Object {
          "node": Object {
            "fileAbsolutePath": "C:/Documents/path-to-the-article-1",
            "frontmatter": Object {
              "cover": Object {
                "childImageSharp": Object {
                  "fluid": Object {
                    "aspectRatio": 1.7,
                    "sizes": "(max-width: 600px) 100vw, 600px",
                    "src": "/static/da8d0fc0af7f0d310517048e303b3fcf/a5a41/preview.jpg",
                    "srcSet": "/static/da8d0fc0af7f0d310517048e303b3fcf/a5a41/preview.jpg",
                  },
                },
              },
              "excerpt": "an excertp for the article #1",
              "path": "path-to-the-article-1",
              "tags": Array [
                "angular",
                "javascript",
                "laravel",
              ],
              "title": "Title of article #1",
            },
          },
        },
        Object {
          "node": Object {
            "fileAbsolutePath": "C:/Documents/path-to-the-article-2",
            "frontmatter": Object {
              "cover": Object {
                "childImageSharp": Object {
                  "fluid": Object {
                    "aspectRatio": 1.7,
                    "sizes": "(max-width: 600px) 100vw, 600px",
                    "src": "/static/da8d0fc0af7f0d310517048e303b3fcf/a5a41/preview.jpg",
                    "srcSet": "/static/da8d0fc0af7f0d310517048e303b3fcf/a5a41/preview.jpg",
                  },
                },
              },
              "excerpt": "an excertp for the article #2",
              "path": "path-to-the-article-2",
              "tags": Array [
                "nodejs",
                "angular",
              ],
              "title": "Title of article #2",
            },
          },
        },
      ]
    }
  />
</div>
`;
